pkg/adapters/opsgenie/oncall/oncall.go

```diff
@@ -0,0 +1,96 @@
+package oncall
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/opsgenie/opsgenie-go-sdk-v2/client"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/schedule"
+	"github.com/ovotech/go-sync/internal/types"
+	"github.com/ovotech/go-sync/pkg/ports"
+)
+
+// Ensure the adapter type fully satisfies the ports.Adapter interface.
+var _ ports.Adapter = &OnCall{}
+
+// ErrNotImplemented is returned if this adapter is used as a destination.
+var ErrNotImplemented = errors.New("not implemented - on-call is readonly")
+
+type iOpsgenieSchedule interface {
+	GetOnCalls(context context.Context, request *schedule.GetOnCallsRequest) (*schedule.GetOnCallsResult, error)
+}
+
+type realClock struct{}
+
+func (realClock) Now() time.Time {
+	return time.Now()
+}
+
+type OnCall struct {
+	client     iOpsgenieSchedule
+	scheduleID string
+	clock      types.Clock
+	logger     types.Logger
+}
+
+// New instantiates a new Opsgenie OnCall adapter.
+func New(opsgenieConfig *client.Config, scheduleID string, optsFn ...func(schedule *OnCall)) *OnCall {
+	logger := log.New(os.Stderr, "[go-sync/opsgenie/oncall]", log.LstdFlags|log.Lshortfile|log.Lmsgprefix)
+	scheduleClient, err := schedule.NewClient(opsgenieConfig)
+
+	if err != nil {
+		logger.Fatalf("Error occurred when creating on-call client: %s", err)
```

Comment:
Instead of halting here, it'd be better to return `*OnCall, error` and let the user deal with it.

Reply:
For consistency, could you move the logger into the struct declaration further down?

Reply:
Which I presumed is a standard interface so don't want to amend?

Reply:
Nope it's fine to amend it 👍